### PR TITLE
feature(component): Rewrite `Alert`s to use themes, resolves #126

### LINKED ISSUE #126: Theme: Alert component
**Is your feature request related to a problem? Please describe.**
Improve the theme support to the component (and subcomponents).

**Describe the solution you'd like**
- All default styles must be moved to the default theme;

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by tulup-conner**
## Breaking changes

None.

## Features

- [x] [Explicitly type <Flowbite theme={}>](https://github.com/themesberg/flowbite-react/pull/151/commits/433f2925f8328dbd4ee6a1fa6cc703aa6425546f)
- [x] [Add Accordion theme defaults](https://github.com/themesberg/flowbite-react/commit/af30327cb0b6a99ee9527b31f9ad9c0328b2ebfc)
- [x] [Rewrite](https://github.com/themesberg/flowbite-react/pull/151/commits/de4e6408ea3f6652b7c4e4ea320eb74e49c21cec) [Alert](https://github.com/themesberg/flowbite-react/pull/151/commits/de4e6408ea3f6652b7c4e4ea320eb74e49c21cec)[s to use themes,](https://github.com/themesberg/flowbite-react/pull/151/commits/de4e6408ea3f6652b7c4e4ea320eb74e49c21cec) [resolves](https://github.com/themesberg/flowbite-react/pull/151/commits/de4e6408ea3f6652b7c4e4ea320eb74e49c21cec) https://github.com/themesberg/flowbite-react/issues/126
- [x] [Check type on <Flowbite theme={..}>](https://github.com/themesberg/flowbite-react/pull/151/commits/b756acd3db323f8a1c0d0899ec019e48a621e6ad)
- [x] [Use semantic colors in <Flowbite theme>](https://github.com/themesberg/flowbite-react/pull/151/commits/a1b48addf128f9f904d30a334667d085b2e5c097)

## Bug fixes

- [x] [Remove redundant a11y label from Alert close button](https://github.com/themesberg/flowbite-react/pull/151/commits/649143bb444d5bb4089b82387509d432da780704)

### CODE REVIEW

**Comment by tulup-conner** [comment]
I was able to test this with the following code in my own app:

```js
<Alert color="primary">Test</Alert>
```

It works great 💯

File: src/lib/theme/default.ts [hunk 1]
```diff
@@ -11,4 +11,29 @@ export default {
       base: 'py-5 px-5 last:rounded-b-lg dark:bg-gray-900',
     },
   },
+  alert: {
+    base: 'flex flex-col gap-2 p-4 text-sm',
+    borderAccent: 'border-t-4',
+    closeButton: {
+      base: '-mx-1.5 -my-1.5 ml-auto inline-flex h-8 w-8 rounded-lg p-1.5 focus:ring-2',
+      color: {
```
Comment (rluders) [inline src/lib/theme/default.ts:19]: I'm wondering if we don't want to replace the color names with some descriptive names... Do you know? Something like: `warning`, `info`, `default`, `success`... 'cause if the person decides to change the gree to use some red-ish color, kind of make it a little bit confusing.

File: src/lib/components/Flowbite/ThemeContext.tsx [hunk 2]
```diff
@@ -6,11 +6,41 @@ import defaultTheme from '../../theme/default';
 export type Mode = string | undefined | 'light' | 'dark';
 
 interface ThemeContextProps {
-  theme: any;
+  theme: FlowbiteTheme;
   mode?: Mode;
   toggleMode?: () => void | null;
 }
 
+export interface FlowbiteTheme {
```
Comment (rluders) [inline src/lib/components/Flowbite/ThemeContext.tsx:14]: THANK YOU! But... may we put it into a separate file? 'Cause, it going to start to be huge. Maybe `ThemeInterface.ts`

**Review by rluders — CHANGES_REQUESTED** [review]
Looks great, just a few comments.

File: src/lib/theme/default.ts (hunk 1 above)
Comment (tulup-conner) [reply src/lib/theme/default.ts:19]: Yes! I think that would make the API a bit more intuitive as well

**Comment by tulup-conner** [comment]
@rluders Using `DeepPartial` you can get full type checking on your custom theme:

```js
const theme: DeepPartial<FlowbiteTheme> = {
    alert: {
      color: {
        primary: "bg-primary",
        // Now provides suggestions while typing in VS Code
      },
    },
  };

  return (
    <Flowbite theme={{ theme }}>
      <Outlet />
    </Flowbite>
  );
```

**Pull request merged**